API.md

```diff
@@ -36,13 +36,13 @@ And a CODAP specific API that was also used for the initial version of the [Clou
 
 The second version of the API is handled by the documents_v2 controller and exposes only a non-GUI programmatic API in the /v2/ namespace as:
 
-- GET /v2/documents/:id - returns the contents of a document specified by id.  This requires either the document's read-only or read-write access keys.
+- GET /v2/documents/:id - returns the contents of a document specified by id.  If the document is not shared this requires either the document's read-only or read-write access keys.
 
 - PUT /v2/documents/:id - updates the contents of a document specified by id.  This requires the document's read-write access keys.
 
 - PATCH /v2/documents/:id - updates the contents of a document specified by id using the JSON Patch specification.  This requires the document's read-write access keys.
 
-- POST /v2/documents - creates a copy of a document identified by the source query parameter and returns the new id of the document along with its read-only and read-write access keys.  The source document must be a shared document or else the copy with fail.  No access key is needed.
+- POST /v2/documents?title=<title>&[shared=(true|false)]&[source=<id>] - either creates a new document using the raw post content or creates a copy of a document identified by the source query parameter.  By default the newly created document does not have the sharing bit set but that can be set by passing ```shared=true``` as a parameter.  If the source parameter is passed it must be a shared document or else the copy with fail.  No access key is needed.  If a document is created this returns a 201 status code with the following json response:  ```{status: "[Created|Copied]", valid: true, id: <document.id>, readAccessKey: <document.read_access_key>, readWriteAccessKey: <document.read_write_access_key>}```.
```

Comment:
If the aim is to replace the v1 api then it might make sense to add the
title to the v2 save endpoint.

- Doug

On Mon, Aug 29, 2016 at 11:20 AM, Scott Cytacki <notifications@github.com>
wrote:

> In API.md
> <https://github.com/concord-consortium/document-store/pull/8#discussion_r76626083>
> :
>
> >
> >  - PUT /v2/documents/:id - updates the contents of a document specified by id.  This requires the document's read-write access keys.
> >
> >  - PATCH /v2/documents/:id - updates the contents of a document specified by id using the JSON Patch specification.  This requires the document's read-write access keys.
> >
> > -- POST /v2/documents - creates a copy of a document identified by the source query parameter and returns the new id of the document along with its read-only and read-write access keys.  The source document must be a shared document or else the copy with fail.  No access key is needed.
> > +- POST /v2/documents?title=<title>&[shared=(true|false)]&[source=<id>] - either creates a new document using the raw post content or creates a copy of a document identified by the source query parameter.  By default the newly created document does not have the sharing bit set but that can be set by passing ```shared=true``` as a parameter.  If the source parameter is passed it must be a shared document or else the copy with fail.  No access key is needed.  If a document is created this returns a 201 status code with the following json response:  ```{status: "[Created|Copied]", valid: true, id: <document.id>, readAccessKey: <document.read_access_key>, readWriteAccessKey: <document.read_write_access_key>}```.
>
> Can we drop the title altogether here? It is not used when the source
> param is set. And it doesn't seem necessary when making make a brand new
> document with the new API.
>
> I don't see anywhere that the title field is returned through the V2 API.
> So the title shouldn't mater. All I can think of is if someone is looking
> at the documents in the document store's UI. Then these title-less
> documents would be annoying.
>
> There doesn't seem to be a way to update the title in the V2 API. So if
> the author changes the title in the CFM and then updates the shared copy,
> the document-store is still going to have the old title. Having the wrong
> title is more confusing than having no title. So either we need to drop the
> title from the V2 API, or we need to support a way to update the title.
> Thoughts? @dougmartin <https://github.com/dougmartin> and @kswenson
> <https://github.com/kswenson>?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/concord-consortium/document-store/pull/8/files/da383d9b0f83ec8670fc6204a51739456a905b0b#r76626083>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAG5KlXC44laEy6xBL5GwkaX4HScoTKyks5qkvi4gaJpZM4JuMxO>
> .
>